src-self-hosted/libc_installation.zig

```diff
@@ -34,6 +38,52 @@ pub const LibCInstallation = struct {
         ZigIsTheCCompiler,
     };
 
+    fn envSubstitution(allocator: *Allocator, input: []const u8, stderr: anytype) ![:0]u8 {
+        var result = try ArrayList(u8).initCapacity(allocator, input.len + 1);
+        errdefer result.deinit();
+
+        var i: usize = 0;
+        outer: while (i < input.len) : (i += 1) {
+            if (input[i] == '$') {
+                for (input[i + 1..]) |c, j| {
```

Comment:
If this doesn't do any loop iterations then we end up with the solitary `$`  being removed.

Probably worth adding a test case for this.